src/Chainweb/Mempool/Mempool.hs

```diff
@@ -145,23 +148,42 @@ import P2P.Peer (PeerInfo)
 
 ------------------------------------------------------------------------------
 data LookupResult t = Missing
-                    | Pending t
+                    | Pending { _lookupResultTx :: !t
```

Comment:
Maybe this record should just become another data type (perhaps `PendingResult`)